packages/react/src/ReactBlockSpec.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  BlockConfig,
+  BlockSpec,
+  createBlockSpec,
+  PropSchema,
+} from "@blocknote/core";
+import { createRoot } from "react-dom/client";
+
+// extend BlockConfig but use a react render function
+export type ReactBlockConfig<
+  Type extends string,
+  PSchema extends PropSchema,
+  ContainsInlineContent extends boolean
+> = Omit<BlockConfig<Type, PSchema, ContainsInlineContent>, "render"> & {
+  render: (props: {
+    block: any;
+    editor: any;
+    contentRef: any;
+  }) => React.ReactNode;
+};
+
+export function createReactBlockSpec<
+  BType extends string,
+  PSchema extends PropSchema,
+  ContainsInlineContent extends boolean
+>(
+  blockConfig: ReactBlockConfig<BType, PSchema, ContainsInlineContent>
+): BlockSpec<BType, PSchema> {
+  const rootEl = document.createElement("div");
+  const root = createRoot(rootEl);
```

Comment:
From testing this code, it seems like it'll only support one instance of this react component at a time? I'm curious why create this element here instead of in the render block.